ts/session/apis/session_backend_server.ts

```diff
@@ -0,0 +1,227 @@
+  }
+
+  // NOTE: returns null if the ed25519 key pair is not found
+  private async getBlindSignedHeaders({
+    method,
+    path,
+    body,
+  }: {
+    method: string;
+    path: string;
+    body: string | null;
+  }): Promise<BlindSignedHeaders | null> {
+    const ed25519SecretKey = (await UserUtils.getUserED25519KeyPairBytes()).privKeyBytes;
+    if (!ed25519SecretKey) {
+      this.logError(`${path} error: ed25519Secret key was not found when creating headers!`);
+      return null;
+    }
+
+    const blindedPkHex = await BlindingActions.blindVersionPubkey({
+      ed25519SecretKey,
+    });
+
+    const sigTimestampSeconds = NetworkTime.nowSeconds();
+    const signedData = await BlindingActions.blindVersionSignRequest({
+      ed25519SecretKey,
+      sigTimestampSeconds,
+      sigMethod: method,
+      sigPath: path,
+      sigBody: body ? new TextEncoder().encode(body) : null,
+    });
+
+    return {
+      'X-FS-Pubkey': blindedPkHex,
+      'X-FS-Timestamp': sigTimestampSeconds.toString(),
+      'X-FS-Signature': fromUInt8ArrayToBase64(signedData),
+    };
+  }
+
+  private async _makeRequest({
+    path,
+    method,
+    bodyGetter,
+    blindSignRequest,
+  }: SessionBackendServerMakeRequestParams): Promise<SessionBackendServerApiResponse> {
+    const body = typeof bodyGetter === 'function' ? await bodyGetter() : null;
+
+    const headers = blindSignRequest
+      ? await this.getBlindSignedHeaders({
+          method,
+          path,
+          body,
+        })
+      : {};
+
+    if (headers === null) {
+      this.logError('failed to blind sign request parameters', path);
+      return SessionBackendServerApi.getGenericErrorResponse();
+    }
+
+    const url = new URL(path, this.server.url);
+
+    const controller = new AbortController();
+    const result = await timeoutWithAbort(
+      OnionSending.sendViaOnionV4ToNonSnodeWithRetries(
+        this.server.xPkHex,
+        url,
+        {
+          method,
+          headers,
+          body,
+          useV4: true,
+        },
+        false,
+        controller.signal,
+        this.server.timeoutMs
+      ),
+      this.server.timeoutMs,
+      controller
+    );
+
+    if (!result || !isOnionV4JSONSnodeResponse(result) || typeof result.body !== 'object') {
+      this.logError(`returned a non json response ${JSON.stringify(result)}`, path);
+      return SessionBackendServerApi.getGenericErrorResponse();
+    }
+
+    if (!batchGlobalIsSuccess(result)) {
+      if (window.sessionFeatureFlags?.debugServerRequests) {
+        this.logError(`failed with status ${parseBatchGlobalStatusCode(result)}`, path);
+      }
+
+      return SessionBackendServerApi.getGenericErrorResponse();
+    }
+
+    return {
+      ...result.body,
+      status_code: result.status_code,
+      t:
+        't' in result.body && typeof result.body.t === 'number'
+          ? result.body.t
+          : NetworkTime.nowSeconds(),
```

Comment:
is this a good idea to fallback to `NetworkTime.nowSeconds(),` here? Should we maybe make it optional instead?